src/RMGConvertLH5.cc

```diff
@@ -292,19 +292,55 @@ bool RMGConvertLH5::ConvertToLH5Internal() {
     LH5Log(RMGLog::error, "not a remage HDF5 output file (invalid header)?");
     return false;
   }
-  LH5Log(RMGLog::detail, "Opened Geant4 HDF5 file ", fHdf5FileName);
+  LH5Log(RMGLog::detail, "opened Geant4 HDF5 file ", fHdf5FileName);
 
   // rework the ntuples to LGDO tables.
   auto ntuples_group = hfile.openGroup(ntuple_group_name);
   auto ntuples = GetChildren(ntuples_group);
   bool ntuple_success = true;
+  std::string links_group_name = "__links__";
+  std::set<std::string> links;
   for (auto& ntuple : ntuples) {
     if (ntuple.empty()) LH5Log(RMGLog::fatal, "empty ntuple name, how is this possible?");
 
     auto det_group = ntuples_group.openGroup(ntuple);
     ntuple_success &= ConvertNTupleToTable(det_group);
     det_group.close();
 
+    // Check fNtupleMeta object for an entry whose second field matches the current ntuple name.
+    for (const auto& item : fNtupleMeta) {
+      // item is a pair: item.first is an int, item.second is a std::pair<int, std::string>
+      if (item.second.second == ntuple) {
+        // create group named "links"
+        if (!ExistsByType(ntuples_group, links_group_name, H5O_TYPE_GROUP)) {
+          auto links_group = ntuples_group.createGroup(links_group_name);
+          links_group.close();
+        }
+
+        // form soft link name "detUID" where UID is item.second.first.
+        auto soft_link_name = "uid" + std::to_string(item.second.first);
+        auto soft_link_name_full = ntuple_group_name + "/" + links_group_name + "/" + soft_link_name;
```

Comment:
I am quite sure that clang-tidy will complain bitterly about this: https://clang.llvm.org/extra/clang-tidy/checks/performance/inefficient-string-concatenation.html